### PR TITLE
rgw: fix bug: delete too many files pre time when bucket check && Optimize the func named check_bad_index_multipart

### PR DESCRIPTION
**Pull request by simon-rock**
1 check_bad_index_multipart assume that all of objs is multipart, we perfrom this step after check if the obj is exist to reduce the msg.
2 fix bug : in the case,  we remove a lot of objs at a time, we will get a errno, -90 (mes is too long),so we  will remove 1000 files per time

### CODE REVIEW

File: src/rgw/rgw_bucket.cc [hunk 1]
```diff
@@ -1310,13 +1310,6 @@ int RGWBucketAdminOp::check_index(RGWRados *store, RGWBucketAdminOpState& op_sta
   Formatter *formatter = flusher.get_formatter();
   flusher.start(0);
 
-  ret = bucket.check_bad_index_multipart(op_state, objs_to_unlink);
```
Comment (simon-rock) [inline src/rgw/rgw_bucket.cc]: check_bad_index_multipart assume that all of objs is multipart, we perfrom this step after check if the obj is exist to reduce the msg.

File: src/rgw/rgw_rados.cc [hunk 2]
```diff
@@ -12257,23 +12257,42 @@ int RGWRados::remove_objs_from_index(RGWBucketInfo& bucket_info, list<rgw_obj_in
...
-  bufferlist updates;
-
-  for (auto iter = oid_list.begin(); iter != oid_list.end(); ++iter) {
-    rgw_bucket_dir_entry entry;
-    entry.key = *iter;
-    dout(2) << "RGWRados::remove_objs_from_index bucket=" << bucket_info.bucket << " obj=" << entry.key.name << ":" << entry.key.instance << dendl;
-    entry.ver.epoch = (uint64_t)-1; // ULLONG_MAX, needed to that objclass doesn't skip out request
-    updates.append(CEPH_RGW_REMOVE | suggest_flag);
-    ::encode(entry, updates);
+  {
```
Comment (simon-rock) [inline src/rgw/rgw_rados.cc]: fix bug : in the case,  we remove a lot of objs at a time, we will get a errno, -90 (mes is too long), will remove 1000 files per time

**Comment by joscollin** [comment]
@simon-rock Your commits are not signed. Could you please fix it ? You can also erase the history and make it a single commit.

**Comment by simon-rock** [comment]
Hi， I have added the sign to this commit.
why is it "Signed-off-by "? I should commit another commit ?
this is the info
commit 9d9730232840a7c36b5cb25036b02c5789c5c211
gpg: Signature made Fri 19 May 2017 02:49:04 PM CST using RSA key ID 19FA7CC8
gpg: Good signature from "simon-rock (simon-rock) <simon29rock@gmail.com>"
Author: simon <ccsi>
Date:   Mon May 8 19:24:21 2017 +0800

    1 check_bad_index_multipart assume that all of objs is multipart, we perfrom this stop after check if the obj is exist to reduce the msg.
    2 fix bug : in the case,  we remove a lot of objs at a time, we will get a errno, -90 (mes is too long), will remove 1000 files per time

    add sign

    update

**Comment by tchaikov** [comment]
@simon-rock it's not about signing the commit with your gpg key. it's a "`Signed-off-by`" line at the end of your commit message? "git commit -s" will do the trick for you. see https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#1-sign-your-work

also you might need to remove the merge commit from your PR.

**Comment by simon-rock** [comment]
make it "signed-off-by"

**Comment by simon-rock** [comment]
@tchaikov 
Is there anything I can do for the status, Unmodified Submodules and make check?
thanks

**Comment by joscollin** [comment]
@simon-rock `git commit -s` will do the trick. 
`git commit --amend` to modify an already made commit.

Please read this too: https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst

**Comment by simon-rock** [comment]
@joscollin  thanks for your return.
Something that confuse me is this:
“Unmodified Submodules — Waiting for status to be reported                  ???
make check — Waiting for status to be reported
Unmodifed Submodules — submodules for project are unmodified          <--
..“

**Comment by joscollin** [comment]
Jenkins retest this please

**Comment by simon-rock** [comment]
It looks like we miss some files about compiling.
Hi @joscollin. can you look at this issue?

**Comment by joscollin** [comment]
@simon-rock The `arm64 make check` doesn't prevent the merge at the moment AFAIK. 

About my previous comment: In Ceph we conventionally put the `Signed-off-by` at the end of a commit message. The title should be prefixed with the submodule name and with a short description of the fix.  This was commented above by Kefu too. Please see one of my merged PR as an example:

`https://github.com/ceph/ceph/pull/15648/commits`
See this too: `https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#1-sign-your-work`

You can do:
1. $git commit --amend.
2. Modify your commit and add the `Signed-off-by` manually for this time.
3. Then push using -f option again: `$git push origin work -f`

From the next time onwards remember to use `-s` when you commit your changes, so that the `Signed-off-by` will be added automatically to your commit, if you configure your git settings properly. 
Please see:` https://help.github.com/articles/setting-your-username-in-git/`
`https://help.github.com/articles/setting-your-email-in-git/`

**Comment by simon-rock** [comment]
@joscollin thanks.
I have change the commit.

**Comment by joscollin** [comment]
Jenkins retest this please

**Comment by simon-rock** [comment]
I think we should put "check_bad_index_multipart"  behind "check_object_index", which will delete the index of the object that dose not exist. so that  will improve the efficiency of the function named “check_bad_index_multipart”

**Comment by simon-rock** [comment]
I think we should put "check_bad_index_multipart"  behind "check_object_index", which will delete the index of the object that dose not exist. so that  will improve the efficiency of the function named “check_bad_index_multipart”

**Pull request closed without merging**